pkg/sentry/kernel/task.go

```diff
@@ -411,7 +411,7 @@ type Task struct {
 
 	// logPrefix is a string containing the task's thread ID in the root PID
 	// namespace, and is prepended to log messages emitted by Task.Infof etc.
-	logPrefix atomic.Value `state:"nosave"`
```

Comment:
I think we should `logPrefix` as-is. Because currently it is storing a string value (not a string pointer). After this change, the string pointer will be stored. IIUC, that will cause the string variable (like `prefix` in the pkg/sentry/kernel/task_log.go:updateInfoLocked()) to also escape to heap (along with the string being heap allocated itself).

Reply:
`atomic.Value` stores an `interface{}` object, which comprises a pointer to `runtime_.type` and a pointer to the represented data (cf. `runtime.eface`), so `atomic.Value` *also* forces heap allocation of the string object (in the caller of `atomic.Value.Store()`, via a call to `runtime.convTstring()`).

Reply:
Ack. Assuming the remaining comments are also moot. The `pgalloc.MemoryFile.mappings` one is on a very hot path, so just want to ensure we are not introducing a regression here.

Reply:
Thanks for the insightful conversation and review! Do you think some more testing is necessary here? If that is the case I could try to look into that :)

Reply:
I think this change is good. I will submit it.